intrahospital_api/management/commands/merge_dup_episode_categories.py

```diff
@@ -0,0 +1,131 @@
+            # singleton, then continue with the more recent data
+            more_recent_data = {}
+            for field in new_singleton._meta.get_fields():
+                field_name = field.name
+                if field_name in merge_patient.IGNORED_FIELDS:
+                    continue
+                more_recent_data[field_name] = getattr(new_singleton, field_name)
+                setattr(new_singleton, field_name, getattr(old_singleton, field_name))
+            new_singleton.save()
+            for field, value in more_recent_data.items():
+                setattr(new_singleton, field, value)
+            with reversion.create_revision():
+                new_singleton.save()
+
+
+def move_non_singletons(subrecord_cls, old_parent, new_parent):
+    """
+    Moves the old_subrecords query set onto the new parent (a patient or episode).
+    In doing so it updates the previous_mrn field to be that of the old_mrn
+    """
+    if new_parent.__class__ == opal_models.Episode:
+        is_episode_subrecord = True
+    else:
+        is_episode_subrecord = False
+    if is_episode_subrecord:
+        old_subrecords = subrecord_cls.objects.filter(episode=old_parent)
+    else:
+        old_subrecords = subrecord_cls.objects.filter(patient=old_parent)
+    for old_subrecord in old_subrecords:
+        if is_episode_subrecord:
+            old_subrecord.episode = new_parent
+        else:
+            old_subrecord.patient = new_parent
+        with reversion.create_revision():
+            old_subrecord.save()
+
+
+def move_record(subrecord_cls, old_parent, new_parent):
+    if getattr(subrecord_cls, "_is_singleton", False):
+        update_singleton(subrecord_cls, old_parent, new_parent)
+    else:
+        move_non_singletons(subrecord_cls, old_parent, new_parent)
+
+
+def merge_episode(*, old_episode, new_episode):
+    for episode_related_model in merge_patient.EPISODE_RELATED_MODELS:
+        move_record(
+            episode_related_model,
+            old_episode,
+            new_episode,
+        )
+
+@transaction.atomic
+def merge_patient_episodes(patient_id):
+    patient = opal_models.Patient.objects.get(id=patient_id)
+    episodes = patient.episode_set.all()
+    category_names = list(set([i.category_name for i in episodes]))
+    for category in category_names:
+        category_episodes = [i for i in episodes if i.category_name==category]
+        if len(category_episodes) > 1:
+            root = category_episodes[0]
+            for category_episode in category_episodes[1:]:
+                merge_patient.update_tagging(category_episode, root)
+                merge_episode(old_episode=category_episode, new_episode=root)
+                category_episode_id = category_episode.id
+                patient_id = category_episode.patient_id
+                category_episode.delete()
+                with open(DELETE_FILE, 'a') as w:
+                    w.write(f'\n{patient_id},{category_episode_id}')
```

Comment:
Same as merge patient, can this be a JSON serialisation of to_dict

Reply:
What do you want to_dicted? Patient or episode?

I've changed it do to episode.to_dict.